internal/packages/packagestructure/content.go

```diff
@@ -0,0 +1,92 @@
+package packagestructure
+
+import (
+	"fmt"
+	"sort"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/yaml"
+
+	corev1alpha1 "package-operator.run/apis/core/v1alpha1"
+	manifestsv1alpha1 "package-operator.run/apis/manifests/v1alpha1"
+	"package-operator.run/package-operator/internal/packages"
+	"package-operator.run/package-operator/internal/packages/packagebytes"
+)
+
+// PackageContent represents the parsed content of an PKO package.
+type PackageContent struct {
+	PackageManifest *manifestsv1alpha1.PackageManifest
+	Manifests       ManifestMap
+}
+
+// Converts the PackageContent back into a FileMap.
+func (pc *PackageContent) ToFileMap() (packagebytes.FileMap, error) {
+	fm, err := pc.Manifests.ToFileMap()
+	if err != nil {
+		return nil, err
+	}
+
+	// ensure GVK is set
+	pc.PackageManifest.SetGroupVersionKind(
+		packages.PackageManifestGroupKind.WithVersion(
+			manifestsv1alpha1.GroupVersion.Version))
+	packageManifestBytes, err := yaml.Marshal(pc.PackageManifest)
+	if err != nil {
+		return nil, fmt.Errorf("marshal YAML: %w", err)
+	}
+	fm[packages.PackageManifestFile] = packageManifestBytes
+	return fm, nil
+}
+
+func (pc *PackageContent) ToTemplateSpec() (templateSpec corev1alpha1.ObjectSetTemplateSpec) {
+	templateSpec.AvailabilityProbes = pc.PackageManifest.Spec.AvailabilityProbes
+
+	objectsByPhase := map[string][]unstructured.Unstructured{}
+	for _, objects := range pc.Manifests {
+		for _, object := range objects {
+			annotations := object.GetAnnotations()
+			phase := annotations[manifestsv1alpha1.PackagePhaseAnnotation]
```

Comment:
Why do we want to remove the phase annotations from the object?

Reply:
I wanted to remove this annotation because it's only significant during package "assembly" and I don't see a reason for persisting it into the cluster.

Reply:
I guess I am wondering if it could ever be useful to have that information there while debugging on the cluster or during an incident